paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/CompactorSourceBuilder.java

```diff
@@ -92,14 +93,27 @@ public CompactorSourceBuilder withPartitions(List<Map<String, String>> partition
                                     .toArray(Predicate[]::new));
         }
 
-        ReadBuilder readBuilder = bucketsTable.newReadBuilder().withFilter(partitionPredicate);
         if (isContinuous) {
+            // set 'streaming-compact' and remove 'scan.bounded.watermark' if exists
+            Options options = Options.fromMap(bucketsTable.options());
+            options.set(CoreOptions.STREAMING_COMPACT, true);
+            Map<String, String> newOptions = options.toMap();
+            if (options.contains(CoreOptions.SCAN_BOUNDED_WATERMARK)) {
+                newOptions.remove(CoreOptions.SCAN_BOUNDED_WATERMARK.key());
```

Comment:
Do you have tests on this?
Dynamic options are new options, here can not just remove `SCAN_BOUNDED_WATERMARK`. Maybe we should pass a `Long.MAX`.

Reply:
`SCAN_BOUNDED_WATERMARK`, `SCAN_TIMESTAMP_MILLIS`, `SCAN_SNAPSHOT_ID` may have conflict with new options when checking.

I think it's more reasonable to pass a map like `scan.bounded.watermark -> null`, then in `Table#copy`, we can remove the `ConfigOption` if passing value is null. WDYT?

Reply:
`null` looks good to me.